app/services/people/creator.rb

```diff
@@ -2,29 +2,34 @@
 
 module People
   class Creator < People::Crud
-    attr_accessor :params, :profile
+    attr_reader :person, :profile
 
     def call
-      create_profile
-    end
-
-    def person
-      profile&.person
+      Profile.transaction do
+        create_person
+        create_profile
+        profile.save!
+      end
     end
 
   private
 
+    def create_person
+      @person = Person.new(
```

Comment:
How do we ensure that this person does not exist before, now we have migrated the new list of fields over? Will we risk creating duplicate profiles if they exist before? (maybe this is done somewhere else I've missed)

Reply:
So this is true before/after this change. I haven't affected the behaviour around uniqueness of either a Profile or Person. It seems like we could do with a dedicated chat/story around our approach to uniqueness of a person/profile. My suspicion is that the frontend workflows handle this currently.

Reply:
spoke in person but for posterity: this still isn't an issue, we will ensure uniqueness in the new endpoints